lib/msal-node/src/cache/CacheManager.ts

```diff
@@ -0,0 +1,94 @@
+import { Storage } from './Storage';
+import { Serializer, Deserializer, InMemoryCache, JsonCache } from '@azure/msal-common';
+import { ICachePlugin } from './ICachePlugin';
+
+const defaultSerializedCache: JsonCache = {
+    Account: {},
+    IdToken: {},
+    AccessToken: {},
+    RefreshToken: {},
+    AppMetadata: {}
+};
+
+
+export class CacheManager {
+
+    hasChanged: boolean;
+    storage: Storage;
+    persistence: ICachePlugin;
+
+    constructor(storage: Storage, cachePlugin?: ICachePlugin) {
+        this.hasChanged = false;
+        this.storage = storage;
+        this.storage.registerChangeEmitter(this.handleChangeEvent.bind(this))
+        if (cachePlugin) {
+            this.setPersistence(cachePlugin);
+        }
+
+    }
+
+    setPersistence(persistence: ICachePlugin): void {
+        this.persistence = persistence;
+    }
+
+    handleChangeEvent() {
+        this.hasChanged = true;
+    }
+
+    mergeState(oldState: JsonCache, currentState: JsonCache) {
+        // TODO
+        // mergeUpdates(old, new)
+        // mergeRemovals(old, new)
+        return {
+            ...oldState,
+            ...currentState
```

Comment:
This does a shallow merge, is that okay?

Reply:
Yeah in the long run this is going to be deep when mergeRemovals and Updates is added.
I may add that in this pr to just get in in